libenv/unix_iface.c

```diff
@@ -657,9 +659,23 @@ static void FindV6InterfacesInfo(EvalContext *ctx)
 
                 if ((IsIPV6Address(ip->name)) && ((strcmp(ip->name, "::1") != 0)))
                 {
+                    /* 8 groups of 4 characters + 7 colons as separators + the prefix (incl. the NUL byte)*/
+                    char prefixed_ip[PREFIXED_IPV6_LEN] = {0};
+                    size_t ret;
                     Log(LOG_LEVEL_VERBOSE, "Found IPv6 address %s", ip->name);
                     EvalContextAddIpAddress(ctx, ip->name, NULL); // interface unknown
                     EvalContextClassPutHard(ctx, ip->name, "inventory,attribute_name=none,source=agent");
+
+                    strcpy(prefixed_ip, "ipv6_");
+                    ret = strlcat(prefixed_ip + (sizeof(IPV6_PREFIX) - 1), ip->name, PREFIXED_IPV6_LEN);
+                    if (ret <= PREFIXED_IPV6_LEN)
+                    {
+                        EvalContextClassPutHard(ctx, prefixed_ip, "inventory,attribute_name=none,source=agent");
+                    }
+                    else
+                    {
+                        Log(LOG_LEVEL_WARNING, "Failed to prefix IPv6 address  %s", ip->name);
```

Comment:
Use `:` delimiter before the final `%s`. And do you expect this ever happening? I believe the buffer should never overflow, in which case a simple `assert()` after `strlcat()` should be enough.

BTW you might want to have a look at `StrCat()`. It's supposed to be a faster alternative of `strlcat()` for when you know all lengths. Not sure if it makes sense to use it here.

Reply:
Yeah, ``assert()`` should be enough here. I replaced the multiple steps with a single ``snprintf()`` call, it's just easier to read and the efficiency should be the same.